### ISSUE TITLE
More variants of 'todo' in comment should be accepted

### ISSUE THREAD

**Issue by mnojek** [MarketSquare/robotframework-robocop#378]
Consider following code snippet:
```
*** Settings ***
Documentation  Suite doc
# TODO found


*** Keywords ***  # TODO not found
Different Place Todos    # TODO found
    [Documentation]  Some doc  # TODO not found
    No Operation
    [Return]        somevalue       # TODO not found
# TODO found

Different Case Todos
    [Documentation]  Some doc
    No Operation    # TODO found
    No Operation    # todo found
    No Operation    # TODO: found
    No Operation    # todo: found
    No Operation    # to-do not found
    No Operation    # TO-DO not found
    No Operation    # to do not found
    No Operation    # TO DO not found

```
I think we need to decide where we should also detect 'todo' in comment and also what variants of it we want to accept. What do you think, @bhirsz ?

**Comment by bhirsz** [comment]
Regarding variants, using PyCharm as reference:
![image](https://user-images.githubusercontent.com/8532066/127472859-85d6c67d-4f9b-4d00-9dd5-62a5263ab331.png)
I think anything that have "todo" (case insensitive) should be detected. `to-do` or `to do` should not be detected.

In some of your examples comments where not processed - it would be good to check our test data with other comment rules and update our checkers to match comments everywhere, ie:
```
*** Keywords ***  # TODO not found -> should be found
Different Place Todos    # TODO found
    [Documentation]  Some doc  # TODO not found -> should be found
```

**Comment by mnojek** [comment]
I decided not to add 'to-do' variant, since it's not commonly used across different IDEs and languages.
I added missing cases and support for them.
